### ISSUE TITLE
Unable to create variations from iOS app

### ISSUE THREAD

**Issue by samy** [woocommerce/woocommerce-ios#14033]
**Describe the bug**
When I try to create variations on variable product, only one variation is created (and no attribute is displayed in intermediate screen)

**To Reproduce**
1. Click on "+" icon
2. Choose "Variable product"
3. Type down random title for the product
4. Click on "Add variations"
5. Click on "Add Attributes"
6. Choose any attribute
7. Choose two values in the list
8. Click on "Next"
9. No attribute is listed here (but I selected one on step 6)
10. Click on "Next"
11. Click on "Generate variation"
12. Click on "Generate all variations"
13. Click on "OK"
14. Only one variation is listed in "Variations" section

**Screenshots**
Video below shows the symptom
https://github.com/user-attachments/assets/a606d1a3-0f0c-40f0-893d-a72f13359953

**Expected behavior**
All variations using all selected attributes are generated

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: iPhone 13
- iOS version: 17.6.1
- WooCommerce iOS version: 20.4.0.1

**WordPress Environment**
<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.3.2
Legacy REST API Package Version: L’extension Legacy REST API n’est pas installée sur ce site.
Action Scheduler Version: ✔ 3.8.1
Log Directory Writable: ✔
WP Version: 6.6.2
WP Multisite: –
WP Memory Limit: 512 Mo
WP Debug Mode: –
WP Cron: –
Language: fr_FR
External object cache: ✔

### Server Environment ###

Server Info: Apache
PHP Version: 8.2.23
PHP Post Max Size: 512 Mo
PHP Time Limit: 360
PHP Max Input Vars: 250000
cURL Version: 7.87.0
OpenSSL/1.1.1w

SUHOSIN Installed: –
MySQL Version: 10.6.19-MariaDB
Max Upload Size: 512 Mo
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

acf-field: 5
acf-field-group: 4
attachment: 20024
custom_css: 3
is_search_form: 1
lieu: 3
nav_menu_item: 26
oembed_cache: 3
page: 9
post: 1223
product: 543
product_variation: 337
revision: 445
shop_coupon: 7
shop_order: 326
shop_order_refund: 6
wp_global_styles: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (27) ###

Advanced Custom Fields PRO: par WP Engine – 6.3.6
AfterShip Tracking - All-In-One WooCommerce Order Tracking (Free plan available): par AfterShip – 1.17.15
Akismet Anti-spam: Spam Protection: par Automattic - Équipe Anti-spam – 5.3.3
Antispam Bee: par pluginkollektiv – 2.11.6
APCu Manager: par Pierre Lannoy / PerfOps One – 4.0.0
Colissimo shipping methods for WooCommerce: par Colissimo – 2.1.0
Custom Post Type UI: par WebDevStudios – 1.17.1
Facebook for WooCommerce: par Facebook – 3.2.9
GA Google Analytics: par Jeff Starr – 20240820
Hide Categories and Products for Woocommerce: par N.O.U.S. Ouvert Utile et Simple – 1.2.9
Jetpack: par Automattic – 13.8.1
Kirki Customizer Framework: par Themeum – 5.1.0
Loco Translate: par Tim Whitlock – 2.6.11
PixLikes: par pixelgrade – 1.1.3
SecuPress Free — Sécurité WordPress: par SecuPress – 2.2.5.3
Top Bar PRO: par WP Darko – 3.0.1
W3 Total Cache: par BoldGrid – 2.7.5
Chronopost & Mondial relay pour WooCommerce - WCMultiShipping: par Mondial Relay WooCommerce - WCMultiShipping – 2.5.2
Classes CSS Widget: par Jory Hogeveen – 1.5.4.1
Widgets in Menu for WordPress: par saurabhshukla
yapapaya – 1.0.0

WooCommerce.com Update Manager: par Automattic – 1.0.3
WooCommerce Stripe Gateway: par WooCommerce – 8.7.0
WooCommerce PayPal Payments: par WooCommerce – 2.9.0
WooCommerce: par Automattic – 9.3.2
Yoast SEO: par L’équipe Yoast – 23.5
WP Crontrol: par John Blackbourn – 1.17.0
WP Product Feed Manager: par WP Marketing Robot – 2.10.0

### Inactive Plugins (1) ###

Smash Balloon Instagram Feed: par Smash Balloon – 6.5.1

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
db.php: db.php
object-cache.php: object-cache.php

### Must Use Plugins (2) ###

SecuPress COOKIEHASH: par  – 2.0
SecuPress Salt Keys: par  – 2.0

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right
Thousand Separator: 
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔
Enforce Approved Product Download Directories: –
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: ✔

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 jours
Level threshold: –
Log directory size: 2 Mo

### WC Pages ###

Base de la boutique: #95253 - /boutique/
Panier: #92435 - /panier/ -  Contient le code court [woocommerce_cart]
Commander: #92436 - /paiement/ -  Contient le code court [woocommerce_checkout]
Mon compte: #92437 - /mon-compte/
Conditions générales de vente et d’utilisation: ❌ Page non définie

### Theme ###

Name: Lulu la lucette
Version: 1.0.0
Author URL: http://example.com
Child Theme: ✔
Parent Theme Name: Kale
Parent Theme Version: 2.7
Parent Theme Author URL: https://www.lyrathemes.com/
WooCommerce Support: ✔

### Templates ###

Archive Template: Votre thème a un fichier woocommerce.php
vous ne serez pas en mesure de substituer le modèle personnalisé woocommerce/archive-product.php étant donné que woocommerce.php a la priorité sur archive-product.php. Ceci vise à prévenir les problèmes d’affichage.

Overrides: lululalucette/woocommerce/loop/sale-flash.php

### WooCommerce PayPal Payments ###

Onboarded: ✔
Shop country code: FR
WooCommerce currency supported: ✔
Advanced Card Processing available in country: ✔
Pay Later messaging available in country: ✔
Webhook status: –
PayPal Vault enabled: –
ACDC Vault enabled: –
Logging enabled: –
Reference Transactions: ✔
Used PayPal Checkout plugin: ✔
Subscriptions Mode: Disabled

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
reactify-classic-payments-settings

Daily Cron: ✔ Next scheduled: 2024-09-25 00:56:53 +02:00
Options: ✔
Notes: 140
Onboarding: completed

### Action Scheduler ###

Terminé: 3 226
Oldest: 2024-08-24 21:35:24 +0200
Newest: 2024-09-24 21:30:07 +0200

En attente: 4
Oldest: 2024-09-24 21:33:16 +0200
Newest: 2024-09-25 19:10:04 +0200


### Status report information ###

Generated at: 2024-09-24 21:34:33 +02:00
```
</details>

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by rachelmcr** [comment]
Hi @samy, thanks for reporting this issue! Could you update to the latest version of the app and check to see if everything is working for you now? There was a bug with variations that sounds similar to what you described here, and that was fixed in the 20.5 release (https://github.com/woocommerce/woocommerce-ios/pull/14046).

**Comment by samy** [comment]
> Hi @samy, thanks for reporting this issue! Could you update to the latest version of the app and check to see if everything is working for you now? There was a bug with variations that sounds similar to what you described here, and that was fixed in the 20.5 release (#14046).

I confirm it is now fixed 💪
